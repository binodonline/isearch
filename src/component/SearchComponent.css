.Search {
  align-content: center;
  text-align: center;
  margin: 10px;
}

.Search-button {
  background-color: #1156e2;
  color: white;
  margin: 10px;
}

.Search-grid {
  display: flex;
  justify-content: center;
  align-items: center;
}
