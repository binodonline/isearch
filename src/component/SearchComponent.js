import React, { useState } from "react";
import { useSelector } from "react-redux";
import { getMergedData } from "./state/selector";
import { AgGridColumn, AgGridReact } from "ag-grid-react";

import "./SearchComponent.css";
import "ag-grid-community/dist/styles/ag-grid.css";
import "ag-grid-community/dist/styles/ag-theme-alpine.css";

const SearchComponent = (props) => {
  const getFinalData = useSelector((state) => getMergedData(state));
  const [year, setYear] = useState("");

  const fetchData = () => {
    props.fetchData(year);
  };

  const displayData = () => {
    return (
      <div className="ag-theme-alpine" style={{ height: 800, width: 600 }}>
        <AgGridReact rowData={getFinalData}>
          <AgGridColumn field="countryName" headerName="Country"></AgGridColumn>
          <AgGridColumn field="population" headerName="Total Population" sortable={true}></AgGridColumn>
          <AgGridColumn field="gdpValue" headerName="GDP (current US$)" sortable={true}></AgGridColumn>
        </AgGridReact>
      </div>
    );
  };

  return (
    <div className="Search">
      <input type="text" value={year} onChange={(e) => setYear(e.target.value)}></input>
      <button className="Search-button" onClick={fetchData}>
        Search
      </button>
      <div className="Search-grid">{displayData()}</div>
    </div>
  );
};

export default SearchComponent;
