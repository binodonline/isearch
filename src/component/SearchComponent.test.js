import React from "react";
import { Provider } from "react-redux";
import { render, fireEvent } from "@testing-library/react";
import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import SearchComponent from "./SearchComponent";
import createStore from "../store";

const store = createStore();
Enzyme.configure({ adapter: new Adapter() });

jest.mock("react-redux", () => ({ ...jest.requireActual("react-redux"), useSelector: jest.fn() }));

describe("test search component", () => {
  it("renders search component correctly", () => {
    const wrapper = shallow(<SearchComponent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("fetchData being called on search button click", () => {
    const fetchData = jest.fn();
    const wrapper = render(
      <Provider store={store}>
        <SearchComponent fetchData={fetchData} />
      </Provider>
    );
    const btn = wrapper.getByText("Search");
    fireEvent.click(btn);
    expect(fetchData).toBeCalled();
  });
});
