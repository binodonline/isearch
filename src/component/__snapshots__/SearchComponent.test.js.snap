// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test search component renders search component correctly 1`] = `
<div
  className="Search"
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="Search-button"
    onClick={[Function]}
  >
    Search
  </button>
  <div
    className="Search-grid"
  >
    <div
      className="ag-theme-alpine"
      style={
        Object {
          "height": 800,
          "width": 600,
        }
      }
    >
      <AgGridReact
        disableStaticMarkup={false}
        legacyComponentRendering={false}
        maxComponentCreationTimeMs={1000}
      >
        <AgGridColumn
          field="countryName"
          headerName="Country"
        />
        <AgGridColumn
          field="population"
          headerName="Total Population"
          sortable={true}
        />
        <AgGridColumn
          field="gdpValue"
          headerName="GDP (current US$)"
          sortable={true}
        />
      </AgGridReact>
    </div>
  </div>
</div>
`;
